File: Factorial/Factorial.py
def EasiestFactorial(num: int) -> int:
    factorial = 1
    for i in range(1, num+1):
        factorial *= i
    return factorial

def FactorialRecursion(num: int) -> int:
    if num < 1:
        return 1
    return num * FactorialRecursion(num-1)

def FactorialRecursionOneLine(num: int) -> int:
    return 1 if num < 1 else num * FactorialRecursion(num-1)

if __name__ == '__main__':
    print(EasiestFactorial(5))
    print(FactorialRecursion(5))
    print(FactorialRecursionOneLine(5))

    print(EasiestFactorial(1))
    print(FactorialRecursion(1))
    print(FactorialRecursionOneLine(1))

    print(EasiestFactorial(0))
    print(FactorialRecursion(0))
    print(FactorialRecursionOneLine(0))

File: Factorial/Factorial.go
package main

import "fmt"

func Factorial(num uint) uint64 {
	var factorial uint64 = 1
	var i uint
	for i = 1; i <= num; i++ {
		factorial *= uint64(i)
	}
	return factorial
}

func FactorialRecursion(num uint) uint64 {
	if num < 1 {
		return 1
	}
	return uint64(num) * FactorialRecursion(num-1)
}

func main() {
	fmt.Println(Factorial(5))
	fmt.Println(Factorial(1))
	fmt.Println(Factorial(0))

	fmt.Println(FactorialRecursion(5))
	fmt.Println(FactorialRecursion(1))
	fmt.Println(FactorialRecursion(0))
}
